nav2_costmap_2d/src/clear_costmap_service.cpp

```diff
@@ -200,10 +203,64 @@ void ClearCostmapService::clearLayerRegion(
   costmap->addExtraBounds(ox, oy, ox + width, oy + height);
 }
 
-void ClearCostmapService::clearEntirely()
+void ClearCostmapService::clearEntirely(const std::vector<std::string> & plugins)
 {
-  std::unique_lock<Costmap2D::mutex_t> lock(*(costmap_.getCostmap()->getMutex()));
-  costmap_.resetLayers();
+  if (plugins.empty()) {
+    // Default behavior: clear all layers
+    std::unique_lock<Costmap2D::mutex_t> lock(*(costmap_.getCostmap()->getMutex()));
+    RCLCPP_INFO(logger_, "Clearing all layers in costmap: %s", costmap_.getName().c_str());
+    costmap_.resetLayers();
+  } else {
+    // Clear only specified plugins
+    auto layers = costmap_.getLayeredCostmap()->getPlugins();
+    bool layer_was_cleared = false;
+    for (auto & layer : *layers) {
+      if (shouldClearLayer(layer, plugins)) {
+        if (layer->isClearable()) {
+          RCLCPP_INFO(logger_, "Clearing entire layer: %s", layer->getName().c_str());
+          auto costmap_layer = std::static_pointer_cast<CostmapLayer>(layer);
+          costmap_layer->resetMap(0, 0, costmap_layer->getSizeInCellsX(),
+            costmap_layer->getSizeInCellsY());
+          layer_was_cleared = true;
+        } else {
+          RCLCPP_WARN(
+            logger_,
+            "Layer '%s' is not clearable, skipping.",
+            layer->getName().c_str());
+        }
+      }
+    }
+    if (layer_was_cleared) {
+      RCLCPP_INFO(logger_, "Resetting master costmap after plugin clearing");
+      costmap_.getCostmap()->resetMap(0, 0,
+        costmap_.getCostmap()->getSizeInCellsX(),
+        costmap_.getCostmap()->getSizeInCellsY());
+      costmap_.updateMap();
```

Comment:
We shouldn't be updating the map in the clearing service, no other clearing service does this